src/back/spv/mod.rs

```diff
@@ -385,8 +385,18 @@ pub struct Writer {
     physical_layout: PhysicalLayout,
     logical_layout: LogicalLayout,
     id_gen: IdGenerator,
-    capabilities: crate::FastHashSet<Capability>,
-    forbidden_caps: Option<&'static [Capability]>,
+
+    /// The set of capabilities modules are permitted to use.
+    ///
+    /// This is initialized from `Options::capabilities`.
+    capabilities_available: Option<crate::FastHashSet<Capability>>,
+
+    /// The set of capabilities used by this module.
+    ///
+    /// If `capabilities_available` is `Some`, then this is always a subset of
```

Comment:
what's the motivation for collecting this subset if `capabilities_available` is provided?

Reply:
We use it to emit `OpCapability` instructions.

Reply:
I don't think it helps in any way versus just passing `capabilities_available` (the superset).